### PR TITLE
[NPU] Support MegaMoE FuseEP mode for Kimi-K3

### PR DESCRIPTION
**Pull request by qybnb**
## Motivation

Enable the Ascend MegaMoE fused dispatch/GEMM/combine path for ModelSlim W4A8 Kimi-K3. The existing NPU FuseEP modes cover the W8A8 paths, while the validated W4A8 deployment needs a separate MegaMoE weight ABI and operator backend.

## Modifications

- Add `--fuseep-mode 3` and keep the deprecated environment-variable compatibility in sync.
- Route mode 3 through `deep_ep.Buffer.fused_deep_moe` with `backend="mega_moe"` and the Kimi-K3 SiTU activation parameters.
- Convert ModelSlim W4A8 expert weights, scales, and biases into the per-expert MegaMoE ABI.
- Share FuseEP weight processing between the W8A8 and W4A8 MoE quantization methods.
- Add NPU unit coverage for CLI parsing, mode-3 operator routing, and W4A8 weight conversion.

## Accuracy Tests

- Targeted mocked NPU unit test on this rebased commit: `3 passed`.
- Four-node Kimi-K3 validation of the mode-3 implementation: GPQA Diamond `94.95%` (`188/198`).

### 8K input / 1K output, no prefix cache

Four-node Kimi K3, concurrency 1, one request, with the same serving parameters except for the MoE backend:

| MoE path | TTFT |
| --- | ---: |
| Without MegaMoE FuseEP mode 3 | `7820.47 ms` |
| MegaMoE FuseEP mode 3 | `4423.65 ms` |
image

MegaMoE FuseEP mode 3 reduces TTFT by **43.44%** (`3396.82 ms`), equivalent to a **1.77x** TTFT speedup.

### 128K input / 90% prefix cache / 1K output

Four-node Kimi K3, concurrency 1, with a verified 115,200-token prefix-cache hit:
- TTFT: `9952.8 ms`
- TPOT: `38.4 ms`
- E2EL: `48300.8 ms`
- Output throughput: `20.7036 token/s`

## Checklist

- [ ] Format your code according to the [Format code with pre-commit](https://docs.sglang.io/developer_guide/contribution_guide.html#format-code-with-pre-commit).
- [x] Add unit tests according to the [Run and add unit tests](https://docs.sglang.io/developer_guide/contribution_guide.html#run-and-add-unit-tests).
- [x] Update documentation according to [Write documentations](https://docs.sglang.io/developer_guide/contribution_guide.html#write-documentations). (The new mode is documented in the CLI help; no standalone documentation page is needed.)
- [x] Provide accuracy and speed benchmark results according to [Test the accuracy](https://docs.sglang.io/developer_guide/contribution_guide.html#test-the-accuracy) and [Benchmark the speed](https://docs.sglang.io/developer_guide/contribution_guide.html#benchmark-the-speed).
- [x] Follow the SGLang code style [guidance](https://docs.sglang.io/developer_guide/contribution_guide.html#code-style-guidance).

---
### CI States

Latest PR Test (Base): :x: [Run #35567955052](https://github.com/sgl-project/sglang/actions/runs/35567955052)
Latest PR Test (Extra): :x: [Run #35567954826](https://github.com/sgl-project/sglang/actions/runs/35567954826)
Latest PR Test (AMD ROCm 10): :x: [Run #35567954996](https://github.com/sgl-project/sglang/actions/runs/35567954996)

### CODE REVIEW

File: python/sglang/srt/hardware_backend/npu/quantization/moe_methods.py [hunk 1]
```diff
@@ -489,34 +349,11 @@ def __init__(self):
...
-            process_fuseep_weights(layer, prefix)
-        layer._fuseep_weights_processed = True
-        return True
-
     def process_weights_after_loading(
         self, layer: torch.nn.Module, weight_prefix: str
     ) -> None:
         # If the FuseEP weight layout is used, process weights via
-        # maybe_apply_fuseep_weights and skip the rest of this method.
+        # maybe_process_fuseep_weights and skip the rest of this method.
```
Comment (qybnb) [inline python/sglang/srt/hardware_backend/npu/quantization/moe_methods.py]: recover note

File: python/sglang/srt/hardware_backend/npu/quantization/moe_methods.py [hunk 2]
```diff
@@ -184,170 +183,31 @@ def _get_bias_args(
...
-            quant_info,
-            weight_prefix,
-            hidden_states,
-            expert_tokens.to(torch.int64),
-            output_dtype,
-            group_list_type=group_list_type,
-            transposed=True,
-            **scale_args,
-        )
+        if get_exec().moe.fuseep_mode == 3:
```
Comment (qybnb) [inline python/sglang/srt/hardware_backend/npu/quantization/moe_methods.py]: can be merged?

File: python/sglang/srt/hardware_backend/npu/quantization/moe_methods.py [hunk 3]
```diff
@@ -184,170 +183,31 @@ def _get_bias_args(
...
-    def __init__(self):
-        super().__init__(quant_config=None)
-        self.matmul = GroupedMatmul()
-        fp4_dtype = _get_float4_e2m1fn_x2_dtype()
-        if fp4_dtype is None:
-            raise RuntimeError("NPU W4A4 MXFP4 MoE requires float4 support.")
-        self.hidden_states_quantizer = HiddenStatesDynamicQuant(
-            quant_dtype=fp4_dtype,
-            use_mx_quant=True,
+        from sglang.srt.hardware_backend.npu.moe.fuseep import (
```
Comment (qybnb) [inline python/sglang/srt/hardware_backend/npu/quantization/moe_methods.py]: recover note

File: python/sglang/srt/hardware_backend/npu/moe/fuseep.py [hunk 4]
```diff
@@ -124,9 +153,13 @@ def _scale_from_float_to_int64(scale: torch.Tensor) -> torch.nn.Parameter:
 def process_fuseep_weights(layer: torch.nn.Module, weight_prefix: str) -> None:
     """Apply the Ascend FuseEP-specific weight layout for a single weight group.
 
-    Invoked by ``maybe_apply_fuseep_weights`` for both ``"w13"`` and ``"w2"``.
+    Invoked by ``maybe_process_fuseep_weights`` for both ``"w13"`` and ``"w2"``.
```
Comment (qybnb) [inline python/sglang/srt/hardware_backend/npu/moe/fuseep.py]: recover note
